modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -292,15 +292,20 @@ object ConsensusStateUpdater {
       proposals.foldMap(a => Map(a -> 1)).toList.map(_.swap).maximumOption.map(_._2)
 
     private def calculateFacilitators(
-      proposed: Set[PeerId],
+      proposed: Map[PeerId, Set[PeerId]],
       local: Set[PeerId],
       removed: Set[PeerId]
     ): Set[PeerId] = {
-      val allProposed = proposed.union(local)
-      seedlist
-        .map(_.intersect(allProposed))
-        .getOrElse(allProposed)
-        .diff(removed)
+      val allProposed = proposed.view.filterKeys(local.contains).values.flatten.toSet.union(local)
```

Comment:
It's no longer `allProposed` but  possibly `nextProposed` (one step deeper in the graph).

Reply:
It never was `allProposed` actually as it was all proposed so far (based on the facilities we got so far). I can change it to `nextProposed`.

Reply:
But naming will never be perfect unfortunately.